packages/blade/src/utils/makeBorderSize/makeBorderSize.ts

```diff
@@ -0,0 +1,10 @@
+function makeBorderSize<T extends number>(s: T): `${T}px`;
+function makeBorderSize<T extends string>(s: T): T;
+function makeBorderSize<T extends number | string>(size: T): `${T}px` | T {
```

Comment:
if we are already overloading types above then why are we again typing things inline here? shouldn't it infer since the function name is same

also, with this inline type defintion we might be creating a bug? because this says the input can be `string` or `number` and then output can be `{T}px` or `T` which means 

`makeBorderSize('50%')` ➡️ `50px` is a valid return type which logically is incorrect. 

correct me if I didn't get this correctly?

Reply:
> if we are already overloading types above then why are we again typing things inline here?

You have to provide the overlapping function arguments but you can omit the return type (although eslint will yell)

But it's best to type the return type explicitly here otherwise it won't throw if we do something like this: 

```ts
function makeBorderSize(size: number | string) {
  if (typeof size === 'number') {
    return `${size}whatever`; // <- should throw error, but doesn't. 
    // In runtime the value will be `whatever` but in type level it's showing `px` if you hover over it, which is really bad.
  }
  return size;
}
```

> also, with this inline type defintion we might be creating a bug? because this says the input can be string or number and then output can be {T}px or T which means
> makeBorderSize('50%') ➡️ 50px is a valid return type which logically is incorrect.

No it won't do that, thats the whole point of having overloads and doing a runtime check of `typeof size === 'number'` inside the function, TypeScript is smart to discriminate the return type based on the argument passed either number or string.